Classes/Database/Process/MysqlCommand.php

```diff
@@ -96,14 +103,10 @@ protected function buildDefaultOutputCallback($outputCallback)
         return $outputCallback;
     }
 
-    protected function buildConnectionArguments()
+    private function buildConnectionArguments(): array
     {
-        if (!empty($this->dbConfig['user'])) {
-            $arguments[] = '-u';
-            $arguments[] = $this->dbConfig['user'];
-        }
-        if (!empty($this->dbConfig['password'])) {
-            $arguments[] = '-p' . $this->dbConfig['password'];
+        if ($configFile = $this->createTemporaryMysqlConfigurationFile($this->dbConfig['user'] ?? null, $this->dbConfig['password'] ?? null)) {
```

Comment:
Maybe pass the whole `dbConfig` array instead? Optional handling can then be done in the target method and other options (like `host`) can be taken into account too.